llm_demo/orchestrator/langgraph/langgraph_orchestrator.py

```diff
@@ -46,131 +49,125 @@
+                        )
+                    ]
+                }
+            return {"messages": [response]}
+
+        def agent_should_continue(self, state: UserState) -> Literal["tools", "end"]:
+            """
+            Function to determine which node is called next
+            """
+            messages = state["messages"]
+            last_message = messages[-1]
+            # If the LLM makes a tool call, then we route to the "tools" node
+            if isinstance(last_message, AIMessage) and last_message.tool_calls:
+                return "tools"
+            # Otherwise, we stop (reply to the user)
+            return "end"
+
+        # define a new graph
+        llm_graph = StateGraph(UserState)
+        llm_graph.add_node("agent", RunnableLambda(call_model))
+        llm_graph.add_node("tools", tool_node)
+
+        # set agent node as the first node to call
+        llm_graph.set_entry_point("agent")
+
+        # add edges
+        llm_graph.add_conditional_edges("agent", agent_should_continue, {"end": END})
+        llm_graph.add_edge("tools", "agent")
+
+        checkpointer = MemorySaver()
+        langgraph_app = llm_graph.compile(checkpointer=checkpointer)
+        self._checkpointer = checkpointer
+        self._langgraph_app = langgraph_app
+
     def user_session_exist(self, uuid: str) -> bool:
         return uuid in self._user_sessions
 
     async def user_session_insert_ticket(self, uuid: str, params: str) -> Any:
-        user_session = self.get_user_session(uuid)
-        response = await user_session.insert_ticket(params)
-        return response
-
-    async def check_and_add_confirmations(self, response: Dict[str, Any]):
-        for step in response.get("intermediate_steps") or []:
-            if len(step) > 0:
-                # Find the called tool in the step
-                called_tool = step[0]
-                # Check to see if the agent has made a decision to call Prepare Insert Ticket
-                # This tool is a no-op and requires user confirmation before continuing
-                if called_tool.tool in self.confirmation_needing_tools:
-                    if called_tool.tool == "Insert Ticket":
-                        flight_info = await validate_ticket(
-                            self.client, called_tool.tool_input
-                        )
-                        return {"tool": called_tool.tool, "params": flight_info}
-                    return {"tool": called_tool.tool, "params": called_tool.tool_input}
-        return None
+        raise NotImplementedError("Irrelevant to LangGraph.")
 
     async def user_session_create(self, session: dict[str, Any]):
         """Create and load an agent executor with tools and LLM."""
-        print("Initializing agent..")
+        client = await self.create_client_session()
+        if self._langgraph_app is None:
+            print("Initializing graph..")
```

Comment:
nit: should really switch to logging at some point. Can file a bug for the backlog though